extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/MicrometerProcessor.java

```diff
@@ -84,6 +99,12 @@ UnremovableBeanBuildItem registerAdditionalBeans(CombinedIndexBuildItem indexBui
                 .addBeanClass(CompositeRegistryCreator.class)
                 .addBeanClass(MeterFilterConstraint.class)
                 .addBeanClass(MeterFilterConstraints.class)
+                .addBeanClass(TIMED_ANNOTATION.toString())
```

Comment:
[nit] Just a thought I had from a general perspective and not specific to this PR only, but can we get smarter about registering beans as being unremovable?

For instance, can we check for usage of `@Timed` in application code before marking the classes for it as unremovable?

Maybe the cost of that search isn't worth the small amount of marking classes as unremovable, but I'm curious

Reply:
yea.. I can chew on it

Reply:
If it's speed of search that's a consideration (and assuming this can be some sort of compile-only dependency) I cannot recommend https://github.com/classgraph/classgraph enough. So fast...

Reply:
Moved annotations (and interceptors) to a different set of additional beans that are removable. Everything still works (I added more tests to an IT bucket to make sure)